server/index.js

```diff
@@ -32,22 +32,43 @@ app.use('/api/users', userRoutes)
 app.use('/api/posts', postRoutes)
 app.use('/api/upload', uploadRoute)
 
-mongoose.connect(process.env.MONGO_URI, { useNewUrlParser: true, useUnifiedTopology: true }, () => {
-  console.log("Connected to database")
+const dbConnect = async () => new Promise((resolve) => {
+  mongoose.connect(process.env.MONGO_URI, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    serverSelectionTimeoutMS: 10000,
```

Comment:
The default timeout is I think 30 seconds, which is a very long time to wait for it to fail! Really it should take milliseconds to connect locally, even 10 seconds is probably excessive.